activerecord/lib/active_record/connection_handling.rb

```diff
@@ -45,16 +45,20 @@ module ConnectionHandling
     # The exceptions AdapterNotSpecified, AdapterNotFound and +ArgumentError+
     # may be returned on an error.
     def establish_connection(spec = nil)
+      raise RuntimeError, "Anonymous class is not allowed." unless name
+
       spec     ||= DEFAULT_ENV.call.to_sym
```

Comment:
Having two variables clash name but differ on semantics is confusing. Maybe renaming this to `spec_name` clears it up a bit?